cert/vault_source.go

```diff
@@ -45,20 +50,27 @@ func (s *VaultSource) client() (*api.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := s.setToken(c); err != nil {
+
+	if err := s.setAuth(c); err != nil {
 		return nil, err
 	}
+
 	return c, nil
 }
 
-func (s *VaultSource) setToken(c *api.Client) error {
+func (s *VaultSource) setAuth(c *api.Client) error {
+	firstCall := true
```

Comment:
Right, since this is a local var but then this doesn't make sense. The `defer` function should then read:

```go
defer func() {
    c.SetToken(s.auth.token)
    if s.auth.token != "" {
        s.checkRenewal(c)
    }
    s.mu.Unlock()
}
```